include/navigation/image_database.h

```diff
@@ -374,29 +387,24 @@ class ImageDatabase
          *        extra fields, if used
          */
         template<class... Ts>
-        void record(const Vector3<millimeter_t> &position, degree_t heading,
-                    const cv::Mat &image, Ts &&... extraFieldValues)
+        void record(const Vector3<millimeter_t> &position, degree_t heading, const cv::Mat &image, Ts &&...extraFieldValues)
         {
-            this->addEntry(image, position, heading, { 0, 0, 0 },
-                           std::forward<Ts>(extraFieldValues)...);
+            this->addEntry(image, getFileName, position, heading, { 0, 0, 0 }, std::forward<Ts>(extraFieldValues)...);
         }
 
         void record(const cv::Mat &image, Entry entry)
         {
-            this->addEntry(image, entry);
+            this->addEntry(image, entry, getFileName);
         }
 
-        std::string getCurrentFilenameRoot() const override
-        {
+    private:
+        const std::function<std::string()> getFileName = [this]() {
```

Comment:
This seems a very odd use of ``std::function``. Why not make this a normal function and bind it when you assign it to a ``std::function`` with ``std::bind(&RouteRecorder::getFileName, this);``

Reply:
Ohh good idea. I didn't know about ``std::bind``.